### PR TITLE
extract callback typing to a file.

### PR DESCRIPTION
**Pull request by septs**

### CODE REVIEW

**Comment by septs** [comment]
```patch
-func makePDUHandler(auth gosmpp.Auth, hook func(*Payload)) func(pdu.PDU, bool) {
+func makePDUHandler(auth gosmpp.Auth, hook func(*Payload)) gosmpp.PDUCallback {
 	return func(p pdu.PDU, responded bool) {
 		// .....
 	}
 }
```

```go
settings := gosmpp.TransceiveSettings{
      ReadTimeout: 30 * time.Second,
      OnPDU:       makePDUHandler(auth, runProgram),
}
```

**Comment by linxGnu** [comment]
Please give description about your motivation for changes. Thank you.

**Comment by septs** [comment]
```go
settings := gosmpp.TransceiveSettings{
      OnPDU: .... // here IDE can provide correct type information
}
```

but when used alone

```go
onPDU := gosmpp.PDUCallback(func (...) { ... })
```

semantics and IDE hint are available

File: types.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package gosmpp
+
+import "github.com/linxGnu/gosmpp/pdu"
+
+type PDUCallback func(pdu pdu.PDU, responded bool)
```
Comment (linxGnu) [inline types.go:5]: Please add comment for exported type

**Review by linxGnu — CHANGES_REQUESTED** [review]
LGTM. Just a nits comment. Thank you very much

**Comment by linxGnu** [comment]
Thank you @septs . LGTM

**Pull request merged**